frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_property_edit/type_option_widget_builder/number.dart

```diff
@@ -0,0 +1,158 @@
+import 'package:flowy_infra_ui/flowy_infra_ui.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:appflowy/generated/locale_keys.g.dart';
+import 'package:go_router/go_router.dart';
+
+class NumberTypeOptionMobileWidgetBuilder extends TypeOptionWidgetBuilder {
+  final NumberTypeOptionMobileWidget _widget;
+
+  NumberTypeOptionMobileWidgetBuilder(
+    NumberTypeOptionContext typeOptionContext,
+  ) : _widget = NumberTypeOptionMobileWidget(
+          typeOptionContext: typeOptionContext,
+        );
+
+  @override
+  Widget? build(BuildContext context) {
+    return _widget;
+  }
+}
+
+class NumberTypeOptionMobileWidget extends TypeOptionWidget {
+  final NumberTypeOptionContext typeOptionContext;
+
+  const NumberTypeOptionMobileWidget({
+    required this.typeOptionContext,
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (context) =>
+          NumberTypeOptionBloc(typeOptionContext: typeOptionContext),
+      child: BlocConsumer<NumberTypeOptionBloc, NumberTypeOptionState>(
+        listener: (context, state) =>
+            typeOptionContext.typeOption = state.typeOption,
+        builder: (context, state) {
+          return GestureDetector(
+            child: Row(
+              children: [
+                PropertyTitle(LocaleKeys.grid_field_numberFormat.tr()),
+                const Spacer(),
+                const HSpace(4),
+                Text(
+                  state.typeOption.format.title(),
+                  style: Theme.of(context).textTheme.titleMedium,
+                ),
+                Icon(
+                  Icons.arrow_forward_ios,
+                  color: Theme.of(context).hintColor,
+                ),
+              ],
+            ),
+            onTap: () => showFlowyMobileBottomSheet(
+              context,
+              isScrollControlled: true,
+              title: LocaleKeys.grid_field_numberFormat.tr(),
+              builder: (bottomsheetContext) => GestureDetector(
+                child: NumberFormatList(
+                  onSelected: (format) {
+                    context
+                        .read<NumberTypeOptionBloc>()
+                        .add(NumberTypeOptionEvent.didSelectFormat(format));
+                    bottomsheetContext.pop();
+                  },
+                  selectedFormat: state.typeOption.format,
+                ),
+              ),
+            ),
+          );
+        },
+      ),
+    );
+  }
+}
+
+typedef SelectNumberFormatCallback = Function(NumberFormatPB format);
+
+class NumberFormatList extends StatelessWidget {
+  final SelectNumberFormatCallback onSelected;
+  final NumberFormatPB selectedFormat;
+  const NumberFormatList({
+    required this.selectedFormat,
+    required this.onSelected,
+    Key? key,
+  }) : super(key: key);
```

Comment:
Personal pref + super.key constructor
```suggestion
  const NumberFormatList({
    super.key,
    required this.selectedFormat,
    required this.onSelected,
  });

  final NumberFormatPB selectedFormat;
  final SelectNumberFormatCallback onSelected;
```